reposerver/repository/repository.go

```diff
@@ -564,16 +566,40 @@ func resolveReferencedSources(hasMultipleSources bool, source *v1alpha1.Applicat
 			}
 			return nil, fmt.Errorf("source referenced %q, which is not one of the available sources (%s)", refVar, strings.Join(refKeys, ", "))
 		}
-		if refSourceMapping.Chart != "" {
-			return nil, errors.New("source has a 'chart' field defined, but Helm charts are not yet not supported for 'ref' sources")
+
+		// Allow OCI ref sources with chart field, but still restrict Git ref sources with chart field
+		if refSourceMapping.Chart != "" && !strings.HasPrefix(refSourceMapping.Repo.Repo, "oci://") {
+			return nil, errors.New("source has a 'chart' field defined, but Helm charts are not yet supported for Git 'ref' sources")
+		}
+
+		var normalizedRepoURL string
+		if strings.HasPrefix(refSourceMapping.Repo.Repo, "oci://") {
+			// For OCI repositories, use the URL as-is without normalization
+			normalizedRepoURL = refSourceMapping.Repo.Repo
+		} else {
+			// For Git repositories, normalize the URL
+			normalizedRepoURL = git.NormalizeGitURL(refSourceMapping.Repo.Repo)
```

Comment:
Can we extract the repo URL normalization to a separate function? It will make the code more readable as this logic is used multiple times in the code and will help with unit testing.